src/osha/oira/services/navigation.py

```diff
@@ -46,11 +49,18 @@ def fix_node(self, node):
         # Fix the portal_type
         if "@type" in node:
             node["@type"] = node["@type"].replace("-", ".")
+            if node["@type"] in self.ignored_portal_types:
+                # Completely ignore this node.
+                return
+
+        for key in self.banned_keys:
+            if key in node:
+                del node[key]
 
         # Recurse into children
         if "items" in node:
-            for child in node["items"]:
-                self.fix_node(child)
+            tree = [self.fix_node(child) for child in node["items"]]
+            node["items"] = list(filter(None, tree))
```

Comment:
It seems to me this tree list is just temporary and can be more conveniently replaced with a generator:
```suggestion
            tree = (self.fix_node(child) for child in node["items"])
            node["items"] = list(filter(None, tree))
```
This is a minor thing, anyway it would be good to keep this in mind for the future.